R/nonmem.R

```diff
@@ -610,7 +569,80 @@ rex::register_shortcuts("babelmixr2")
     .extra <- paste0(paste(.prefixLines, collapse="\n"), "\n")
     rxode2::rxAssignControlValue(ui, ".nmPrefixLines", NULL)
   }
-  return(paste0(.extra, .rxToNonmemGetIndent(ui), .var, "=", .val))
+  paste0(
+    .extra,
+    .rxToNonmemGetIndent(ui), .var, "=", .val,
+    babelmixr2Deparse(x) # Show the source where this code came from
+  )
+}
+
+# When there is a more complex left-hand-side assignment (e.g. initial condition
+# setting or lag time setting)
+.rxToNonmemHandleAssignmentOperatorComplexLHS <- function(x, ui) {
+  stopifnot(length(x[[2]]) %in% c(2, 3))
+  if (length(x[[2]]) == 3) {
+    # Is d/dt() the only 3-long option?
+    ret <- .rxToNonmemHandleDdt(x, ui)
+  } else if (identical(x[[2]][[2]], 0)) {
+    # set initial conditions
+    ret <- .rxToNonmemHandleInitialConditions(x, ui)
+  } else {
+    ret <- .rxToNonmemHandleComplexOther(x, ui)
+  }
+  paste0(
+    .rxToNonmemGetIndent(ui),
+    ret,
+    babelmixr2Deparse(x)
+  )
+}
+
+#'  Handle d/dt() expression
+#'
+#' @param x2 numerator expression
+#' @param x3 denominator expression
+#' @param ui rxode2 ui
+#' @return DADT(#) for NONMEM
+#' @author Matthew L. Fidler
+#' @noRd
+.rxToNonmemHandleDdt <- function(x, ui) {
+  # Verify that it is really d/dt(cmt)
+  lhs <- x[[2]]
+  stopifnot(length(lhs) == 3)
+  stopifnot(lhs[[1]] == as.name("/"))
```

Comment:
I typically use `quote(`/`)` but this seems to work too.  It can be vectorized somehow though, could this cause issues elsewhere since it should be atomic and never a vector.

Reply:
Okay.  I'll confirm that it is identical to the atomic value.

Reply:
I've updated this in a commit that is coming momentarily.